book/src/design/proving-system.md

```diff
@@ -1 +1,69 @@
 # Proving system
+
+The Halo 2 proving system can be broken down into five stages:
+
+1. Commit to polynomials encoding the main components of the circuit:
+   - Cell assignments.
+   - Permuted values for each lookup argument.
+   - Equality constraint permutations.
+   - Products for each lookup argument.
+2. Construct the vanishing argument to constrain all circuit relations to zero:
+   - Custom gates.
+   - Lookup argument rules.
+   - Equality constraint permutation rules.
+3. Evaluate the above polynomials at all necessary points:
+   - All relative rotations used by custom gates across all columns.
+   - Vanishing argument pieces.
+4. Construct the multipoint opening argument to check that all evaluations are consistent
+   with their respective commitments.
+5. Run the inner product argument to create a polynomial commitment opening proof for the
+   multipoint opening argument polynomial.
+
+These stages are presented in turn across this section of the book.
+
+## Example
+
+To aid our explanations, we will at times refer to the following example circuit:
```

Comment:
Let's call this a "constraint system" rather than a circuit. The distinction is useful because we will (in general) be trying to _embed_ arithmetic circuits into relations that involve constraint systems.